src/Validator/LdapUsernameValidator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the package t3g/symfony-ldap-bundle.
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace T3G\Bundle\LdapBundle\Validator;
+
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use T3G\Bundle\LdapBundle\Service\LdapService;
+
+class LdapUsernameValidator extends ConstraintValidator
+{
+    /**
+     * @var LdapService
+     */
+    private $ldapService;
+
+    public function __construct(LdapService $ldapService)
+    {
+        $this->ldapService = $ldapService;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
+        /* @var $constraint LdapUsername */
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        try {
+            $this->ldapService->findUserByName($value);
```

Comment:
I think this is problematic because in `findUserByName` all different exceptions are already handled. No matter if there is a connection issue, a non existing user or multiple users with the same name, that method always throws the same exception which means that the validator cannot properly tell the actual issue. Depending on the use case, the exact issue is important to know.